### PR TITLE
USAGOV-2016-v2: Redirect on bad characters in URL

### PR DESCRIPTION
**Pull request by DaleMHFrey**
Redirect when there's a non-breaking space at the end of a request path

https://cm-jira.usa.gov/browse/USAGOV-2016

## Description
Updated the nginx configs to redirect the user away from a URL that ends in a space/thin-space/similar to the same URL without it.
While I was able to test this in the CMS configuration, this is a blind update to the WWW nginx configs, as I am not sure how to test/run the WWW docker container on my local? This test mainly needs to be done on the WWW configs.

## Type of Changes

- [ ] New Feature
- [ ] Bugfix
- [x] Frontend (Twig, Sass, JS)
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] WWW
  - [ ] Egress
  - [ ] Tools
  - [ ] Cron
- [ ] Other

## Testing Instructions
* Going to a URL of `/foo/bar` should load a not-found page, which is fine
* but after merging this, going to `/foo/bar%C2%A0` should redirect you to `/foo/bar`
* and so should `/foo/bar%E2%80%89` and `/foo/bar%E2%80%83`
* mainly I want to make sure that this configuration is now crashing the WWW nginx container on Dev since this is kind of a blind edit there as I dont quite know how to run the WWW container on my local

## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

File: .docker/src-www/etc/nginx/partials/staticsite.conf.tmpl [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+            set $clean_uri $1;
+            return 301 $clean_uri;
+        }
+        # Match and strip other partial %C2 sequences from the end of the URI
+        if ($uri ~* ^(.+?)(%C2|\xC2)+$) {
+            set $clean_uri $1;
+            return 301 $clean_uri;
+        }
+
+        # try_files $uri @rewrite; # For Drupal <= 6
```
Comment (akf) [inline .docker/src-www/etc/nginx/partials/staticsite.conf.tmpl]: We don't need these try_files lines for the static site. There's no Drupal there and should be no index.php. (Looks like just an accidental copy-n-paste.)

File: .docker/src-cms/etc/nginx/partials/drupal.conf.tmpl [hunk 2]
```diff
@@ -79,6 +79,18 @@
     }
 
     location / {
+        # Decode the URI for proper matching
+        set $decoded_uri $uri;
+        if ($decoded_uri ~* ^(.+?)(%C2%A0|\xA0|%E2%80%89|\xE2\x80\x89|%E2%80%83|\xE2\x80\x83)+$) {
```
Comment (akf) [inline .docker/src-cms/etc/nginx/partials/drupal.conf.tmpl:85]: I would like a comment here -- which characters are these?

**Review by akf — CHANGES_REQUESTED** [review]
See comments.

File: .docker/src-cms/etc/nginx/partials/drupal.conf.tmpl (hunk 2 above)
Comment (DaleMHFrey) [reply .docker/src-cms/etc/nginx/partials/drupal.conf.tmpl:85]: Ah good idea, this is now done 👍

File: .docker/src-www/etc/nginx/partials/staticsite.conf.tmpl (hunk 1 above)
Comment (DaleMHFrey) [reply .docker/src-www/etc/nginx/partials/staticsite.conf.tmpl]: oops, you'r right. This is now fixed

**Review by akf — APPROVED** [review]

**Pull request merged**